File: contractstudy/src/test/java/contractstudy/hierarchy/SuperCallSiteExtractor/java/testData/SubClass1.java
package contractstudy.hierarchy.SuperCallSiteExtractor.java.testData;

public class SubClass1 extends ParentClass {

  public SubClass1() {
    super("test");
  }

  @Override
  public String test() {
    super.test();
    return "test";
  }

  @Override
  public String test2(String param1) {
    return param1;
  }

  public int test3(int param1, int param2) {
    int sum = super.testSum(param1, param2);
    return param1 + sum;
  }
}

File: contractstudy/src/test/java/contractstudy/hierarchy/SuperCallSiteExtractor/kotlin/testData/SubClass1.kt
package contractstudy.hierarchy.SuperCallSiteExtractor.kotlin.testData

class SubClass1 : ParentClass {

    constructor(text: String) : super(text)

    override fun test(): String {
        super.test()
        return "test"
    }

    override fun test2(param1: String): String {
        return param1
    }

    fun test3(param1: Int, param2: Int): Int {
        val sum = super.testSum(param1, param2)
        return param1 + sum
    }
}